pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -38,6 +38,85 @@ class PlotDataItem(GraphicsObject):
     sigClicked = QtCore.Signal(object, object)
     sigPointsClicked = QtCore.Signal(object, object, object)
     sigPointsHovered = QtCore.Signal(object, object, object)
+    
+    class PlotDataset(object):
+        """ Holds collected information for a plotable dataset """
+        def __init__(self, x, y):
+            super().__init__()
+            assert x is not None
+            assert y is not None
+            self.x = x
+            self.y = y
+            self._dataRect = None
+            self.containsNonfinite = None
+            
+        # def copy(self):
+        #     """ Provides a shallow copy of this PlotDataset """
+        #     dataset = self.PlotDataSet( self.x, self.y )
+        #     dataset._dataRect = self._dataRect
+        #     dataset.containsNonfinite = self.containsNonfinite
+        #     return dataset
+            
+        def _updateDataRect(self):
+            """ Find bounds of plotable data """
+            if self.y is None or self.x is None:
+                return None
+            selection = np.isfinite(self.y) # We are looking for the bounds of the plottable data set. Infinite and Nan are ignored. 
+            all_y_are_finite = selection.all() # False if all values are finite, True if there are any non-finites
+            try:
+                ymin = np.min( self.y[selection] ) # find minimum of all finite values
+                ymax = np.max( self.y[selection] ) # find maximum of all finite values
+            except ValueError: # is raised then there are no finite values
+                ymin = np.nan
+                ymax = np.nan
+                
+            selection = np.isfinite(self.x) # We are looking for the bounds of the plottable data set. Infinite and Nan are ignored. 
+            all_x_are_finite = selection.all() # False if all values are finite, True if there are any non-finites
+            try:
+                xmin = np.min( self.x[selection] ) # find minimum of all finite values
+                xmax = np.max( self.x[selection] ) # find maximum of all finite values
+            except ValueError: # is reaised then there are no finite values
+                xmin = np.nan
+                xmax = np.nan
+                
+            self._dataRect = QtCore.QRectF( QtCore.QPointF(xmin,ymin), QtCore.QPointF(xmax,ymax) )
+            self.containsNonfinite = not (all_x_are_finite and all_y_are_finite) # This always yields a definite True/False answer
```

Comment:
nit-pick level 1000 here, should it be `containsNonFinite` ?

Reply:
I *think* "nonfinite" is a word, at least as an adjective.
I see a parallel here between 'infinite' // 'nonfinite' and would consider NonFinite to be even less elegant :)

Reply:
Haha, I'm convinced, resolved!